privaterelay/models.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.db import models
+
+from allauth.socialaccount.models import SocialAccount
+
+
+def update_or_create_subscription(social_account: SocialAccount) -> Subscription:
+    subscription = Subscription.objects.get_or_create(user=social_account.user)[0]
+
+    user_subscriptions = social_account.extra_data.get("subscriptions", [])
+    relay_subscriptions = ""
+    for sub in settings.SUBSCRIPTIONS_WITH_UNLIMITED:
+        if sub in user_subscriptions:
+            relay_subscriptions += f"{sub},"
+
+    subscription.names = relay_subscriptions
+    subscription.save()
+    return subscription
+
+
+class Subscription(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    names = models.TextField(default=None, blank=True, null=True)
```

Comment:
I think it will be harder to manage, but this should be a Many-To-Many relationship with the `Profile`. Something like:

```
class Subscription(models.Model):
     name = Models.Charfield(max_length=100, unique=True)
     
....
     
class Profile(models.Model):
    subscriptions = models.ManyToManyField(Subscription)
```

This would allow going the other way - getting the set of all user with a particular subscription. I'd add a Subscription "free" for free users, so you can identify users where the subscription has been set, versus not populated yet, but you could also populate in a data migration and no subscriptions = free.